### PR TITLE
Fix coverity issues

### PR DESCRIPTION
**Pull request by anmyachev**

### CODE REVIEW

File: bin/triton-tensor-layout.cpp [hunk 1]
```diff
@@ -125,13 +125,13 @@ LogicalResult printLayoutFromFile(MLIRContext *context, StringRef filename,
 
   if (names.empty())
     // If no alias name is given, we print all layout attributes in the file.
-    for (auto def : asmState.getAttributeAliasDefs()) {
+    for (const auto &def : asmState.getAttributeAliasDefs()) {
```
Comment (etiotto) [inline bin/triton-tensor-layout.cpp:128]: If you want to make this change we should upstream it to avoid diverging.

Comment (anmyachev) [reply bin/triton-tensor-layout.cpp:128]: @etiotto thanks for the review! Will try to upstream.

File: third_party/intel/lib/Dialect/TritonGEN/IR/TritonGENOps.cpp [hunk 2]
```diff
@@ -253,10 +253,6 @@ verify2DBlockLoadHWRestriction(TritonGEN::Matrix2DBlockLoadOp op) {
         return op.emitOpError("expecting tile_height to be between 1 and 32");
       if (tileWidth < 1 || tileWidth > 8)
         return op.emitOpError("expecting tile_width to be between 1 and 8");
-      if (tileWidth * vBlocks > 8)
```
Comment (anmyachev) [inline third_party/intel/lib/Dialect/TritonGEN/IR/TritonGENOps.cpp:256]: Dead code; `vBlocks` is always `1` here.

File: third_party/intel/lib/TritonIntelGPUToLLVM/PrintOpToLLVM.cpp [hunk 3]
```diff
@@ -201,9 +201,6 @@ struct PrintOpConversion
     std::string prefix = "%";
     if (width.has_value()) {
       prefix += std::to_string(*width);
-    } else if (hex) {
-      prefix += "0";
-      prefix += std::to_string(type.getIntOrFloatBitWidth() / 4);
```
Comment (anmyachev) [inline third_party/intel/lib/TritonIntelGPUToLLVM/PrintOpToLLVM.cpp:206]: Dead code; see line 190

File: bin/triton-tensor-layout.cpp (hunk 1 above)
Comment (anmyachev) [reply bin/triton-tensor-layout.cpp:128]: @etiotto these changes are merged into Triton

Comment (etiotto) [reply bin/triton-tensor-layout.cpp:128]: Fantastic @anmyachev !

File: third_party/intel/lib/Dialect/TritonGEN/IR/TritonGENOps.cpp (hunk 2 above)
Comment (etiotto) [reply third_party/intel/lib/Dialect/TritonGEN/IR/TritonGENOps.cpp:256]: Thanks for the comment, very useful during code review.

**Review by etiotto — APPROVED** [review]

File: third_party/intel/lib/TritonIntelGPUToLLVM/PrintOpToLLVM.cpp (hunk 3 above)
Comment (whitneywhtsang) [reply third_party/intel/lib/TritonIntelGPUToLLVM/PrintOpToLLVM.cpp:206]: can we upstream this change in common `PrintOpToLLVM.cpp` as well?

**Review by whitneywhtsang — COMMENTED** [review]
In general, can we upstream similar changes in the common version as well?

**Comment by anmyachev** [comment]
> In general, can we upstream similar changes in the common version as well? `LinearLayoutConversions.cpp`, `ConvertLayoutOpToLLVM.cpp`, `RewriteTensorPointer.cpp`?

@whitneywhtsang I will try :)

**Pull request merged**